commands/cli/parse.go

```diff
@@ -227,11 +229,61 @@ func parseArgs(inputs []string, stdin *os.File, argDefs []cmds.Argument, recursi
 		numInputs += 1
 	}
 
+	searchUnknownCmd := func(args []string) []string {
+		arg := args[0]
+		var suggestions []string
+		sortableSuggestions := make(suggestionSlice, 0)
+		var sFinal []string
+		const MIN_LEVENSHTEIN = 3
+
+		var options levenshtein.Options = levenshtein.Options{
+			InsCost: 1,
+			DelCost: 3,
+			SubCost: 2,
+			Matches: func(sourceCharacter rune, targetCharacter rune) bool {
+				return sourceCharacter == targetCharacter
+			},
+		}
+
+		// Start with a simple strings.Contains check
+		for name, _ := range root.Subcommands {
+			if strings.Contains(arg, name) {
+				suggestions = append(suggestions, name)
+			}
+		}
+
+		// If the string compare returns a match, return
+		if len(suggestions) > 0 {
+			return suggestions
+		}
+
+		for name, _ := range root.Subcommands {
+			lev := levenshtein.DistanceForStrings([]rune(arg), []rune(name), options)
+			if lev <= MIN_LEVENSHTEIN {
+				sortableSuggestions = append(sortableSuggestions, &suggestion{name, lev})
+			}
+		}
+		sort.Sort(sortableSuggestions)
+
+		for _, j := range sortableSuggestions {
+			sFinal = append(sFinal, j.cmd)
+		}
+		return sFinal
+	}
+
 	// if we have more arg values provided than argument definitions,
 	// and the last arg definition is not variadic (or there are no definitions), return an error
 	notVariadic := len(argDefs) == 0 || !argDefs[len(argDefs)-1].Variadic
 	if notVariadic && len(inputs) > len(argDefs) {
```

Comment:
I wonder if there are any legitimate cases where we would want to print this.. Or if this always means 'you spelled it wrong'.